pkg/login/social/generic_oauth.go

```diff
@@ -170,6 +166,10 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 		}
 	}
 
+	if s.roleAttributeStrict && !userInfo.Role.IsValid() {
```

Comment:
Thinking about this again, this is actually a bug fix right? Before generic oauth would allow empty with strict mode.